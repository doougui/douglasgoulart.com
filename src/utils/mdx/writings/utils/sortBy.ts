import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
