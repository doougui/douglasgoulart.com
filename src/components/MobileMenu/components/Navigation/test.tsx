import { render, screen } from 'utils/tests';

import { Navigation } from '.';

describe('<Navigation />', () => {
  it('should render closed state', () => {
    const { container } = render(<Navigation isOpen={false} />);

    expect(screen.getByTestId('container')).toHaveStyle({
      pointerEvents: 'none',
    });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render open state', () => {
    render(<Navigation isOpen={true} />);

    expect(screen.getByTestId('container')).toHaveStyle({
      pointerEvents: 'auto',
    });
  });
});
