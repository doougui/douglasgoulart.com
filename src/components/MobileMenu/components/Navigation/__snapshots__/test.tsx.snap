// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render closed state 1`] = `
.c2 {
  padding-bottom: 2rem;
  padding-top: 2rem;
  border-bottom: 1px solid #504F57;
}

.c3 {
  color: #EDEDEF;
  font-size: 1.6rem;
  line-height: 1.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  width: 100%;
  list-style: none;
  margin-top: 10rem;
  text-align: right;
}

.c0 {
  width: 100vw;
  height: 100vh;
  background: #151718;
  position: fixed;
  left: 0;
  top: 0;
  z-index: 20;
  padding: 0 calc(4rem / 2);
  pointer-events: none;
}

<nav
  aria-hidden="true"
  class="c0"
  data-testid="container"
>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/"
      >
        Home
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/about"
      >
        About
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/writings"
      >
        Writings
      </a>
    </li>
    <li
      class="c2"
    >
      <a
        class="c3"
        href="/guestbook"
      >
        Guestbook
      </a>
    </li>
  </ul>
</nav>
`;
